gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/event/lineage/LineageEventBuilder.java

```diff
@@ -65,8 +67,8 @@ public LineageEventBuilder(String name) {
   @Override
   public GobblinTrackingEvent build() {
     Map<String, String> dataMap = Maps.newHashMap(metadata);
-    dataMap.put(SOURCE, Descriptor.serialize(source));
-    dataMap.put(DESTINATION, Descriptor.serialize(destination));
+    dataMap.put(SOURCE, GSON.toJson(source));
```

Comment:
Did you consider adding `toJson() / fromJson() / fromJsonList()` methods on `Descriptor` itself? That way callers can't forget to build a GSON object with the `GsonInterfaceAdapter` and incorrectly serialize the class

Reply:
I did think about it. I was hesitating between whether I should decouple the ser/der from the class definition or not.

Let me do that. It enhances SOT of ser/de.

Reply:
Yeah, I always go back and forth - but I figure it's a code smell if you're repeating the same serde logic in many places